### ISSUE TITLE
Router : Type 'MyContext' does not satisfy the constraint 'Context'.

### ISSUE THREAD

**Issue by MrSaeedNasiri** [grammyjs/website#273]
according to:
https://grammy.dev/plugins/router.html#combining-routers-with-sessions

env:

- Deno@V1.20.3
- grammy@v1.7.1
- grammy_router@v1.1.1

**In use router section**
`const router = new Router<MyContext>((ctx) => ctx.session.step);`

error raise top of MyContext : 

`Type 'MyContext' does not satisfy the constraint 'Context'.
  The types of 'api.raw.setWebhook' are incompatible between these types.
    Type '(args: { url: string; certificate?: import("https://deno.land/x/grammy@v1.7.1/platform.deno").InputFile | undefined; ip_address?: string | undefined; max_connections?: number | undefined; allowed_updates?: readonly ("poll" | ... 12 more ... | "chat_join_request")[] | undefined; drop_pending_updates?: boolean | undef...' is not assignable to type '(args: { url: string; certificate?: import("https://deno.land/x/grammy@v1.7.2/platform.deno").InputFile | undefined; ip_address?: string | undefined; max_connections?: number | undefined; allowed_updates?: readonly ("poll" | ... 12 more ... | "chat_join_request")[] | undefined; drop_pending_updates?: boolean | undef...'.
      Types of parameters 'args' and 'args' are incompatible.
        Type '{ url: string; certificate?: import("https://deno.land/x/grammy@v1.7.2/platform.deno").InputFile | undefined; ip_address?: string | undefined; max_connections?: number | undefined; allowed_updates?: readonly ("poll" | ... 12 more ... | "chat_join_request")[] | undefined; drop_pending_updates?: boolean | undefined; }' is not assignable to type '{ url: string; certificate?: import("https://deno.land/x/grammy@v1.7.1/platform.deno").InputFile | undefined; ip_address?: string | undefined; max_connections?: number | undefined; allowed_updates?: readonly ("poll" | ... 12 more ... | "chat_join_request")[] | undefined; drop_pending_updates?: boolean | undefined; }'.
          Types of property 'certificate' are incompatible.
            Type 'import("https://deno.land/x/grammy@v1.7.2/platform.deno").InputFile | undefined' is not assignable to type 'import("https://deno.land/x/grammy@v1.7.1/platform.deno").InputFile | undefined'.
              Property '[toRaw]' is missing in type 'import("https://deno.land/x/grammy@v1.7.2/platform.deno").InputFile' but required in type 'import("https://deno.land/x/grammy@v1.7.1/platform.deno").InputFile'.deno-ts(2344)`

**Comment by KnorpelSenf** [comment]
> **In use router section** const router = new Router((ctx) => ctx.session.step);

This line of code does not exist on the website in the TypeScript section. The documentation is correct.

In order to fix the problem, you will need to pass your custom context type to the `Router` constructor, just as illustrated in the docs.
```ts
const router = new Router<MyContext>((ctx) => ctx.session.step);
```

**Comment by MrSaeedNasiri** [comment]
sorry, i have not used ` around the code section in issue :
```typescript
const router = new Router<MyContext>((ctx) => ctx.session.step);
```
So it was not displayed correctly. ( edited )

I'm double check the code typescript section code work correctly.
but in deno section i have error:
`Type 'MyContext' does not satisfy the constraint 'Context'.`

:white_check_mark:  Fortunately, the problem was found and solved by upgrading grammy from `grammy@v1.7.1` to `grammy@v1.7.2`

**Comment by KnorpelSenf** [comment]
Perfect!
